test/module/irohad/ametsuchi/postgres_executor_test.cpp

```diff
@@ -1996,5 +1996,204 @@ namespace iroha {
+                      account2_id, "key", "value", boost::none),
+                  false,
+                  account_id));
+      auto kv = sql_query->getAccountDetail(account2_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
+    }
+
+    /**
+     * @given command
+     * @when trying to set kv when has role permission
+     * @then kv is set
+     */
+    TEST_F(CompareAndSetAccountDetail, ValidRolePerm) {
+      addAllPerms();
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+                      account2_id, "key", "value", boost::none),
+                  false,
+                  account_id));
+      auto kv = sql_query->getAccountDetail(account2_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
+    }
+
+    /**
+     * @given command
+     * @when trying to set kv while having no permissions
+     * @then corresponding error code is returned
+     */
+    TEST_F(CompareAndSetAccountDetail, NoPerms) {
+      auto cmd_result =
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+                      account2_id, "key", "value", boost::none),
+                  false,
+                  account_id);
+
+      std::vector<std::string> query_args{account2_id, "key", "value"};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 2, query_args);
+
+      auto kv = sql_query->getAccountDetail(account2_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{}");
+    }
+
+    /**
+     * @given command
+     * @when trying to set kv to non-existing account
+     * @then corresponding error code is returned
+     */
+    TEST_F(CompareAndSetAccountDetail, NoAccount) {
+      addAllPerms();
+      auto cmd_result =
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+                      "doge@noaccount", "key", "value", boost::none),
+                  false,
+                  account_id);
+
+      std::vector<std::string> query_args{"doge@noaccount", "key", "value"};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 3, query_args);
+    }
+
+    /**
+     * @given command
+     * @when trying to set kv and then set kv1 with correct old value
+     * @then kv1 is set
+     */
+    TEST_F(CompareAndSetAccountDetail, ValidOldValue) {
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value", boost::none)));
+
+      auto kv = sql_query->getAccountDetail(account_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
+
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id,
+              "key",
+              "value1",
+              boost::optional<
```

Comment:
should be possible to use an implicit constructor
```suggestion
              "value")));
```

Reply:
No, there is a conflict of types
`char[6] cannot be converted to boost::opional<string>`
if call `make_optional` then 
`boost::optional<char[6]> cannot be converted to boost::opitonal<string>`